src/common/adbc/nanoarrow/allocator.cpp

```diff
@@ -0,0 +1,57 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Maybe a question for @Mytherin .

Should nanoarrow live in the third_party folder with a proper license?

Reply:
I spoke with Mark yesterday, he mentioned it might be a better idea to have the ADBC code live in a separate extension.
If we only keep the init function and make the AdbcDriver static, we can initialize it as part of the extension load.
Before it's loaded we can just return an error from the init method

Reply:
That makes sense imo.